x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -584,6 +584,14 @@ protected void doSaveState(IndexerState indexerState, Map<String, Object> positi
                                 next.run();
                             }
                     ));
+
+            // TODO: check whether continuous data frames is enabled when available
+            if (indexerState.equals(IndexerState.STARTED) && transformTask.currentCheckpoint.get() == 1) {
+                auditor.info(transformConfig.getId(), "Stopping data frame transform and removing task");
+                logger.info("Stopping data frame transform [{}] and removing task", transformConfig.getId());
+                transformTask.shutdown();
```

Comment:
The last state saved in the index will be `STARTED` and there are no guarantees that the last state will be persisted by the time we stop this task. It seems to me that this should be moved into the `putOrUpdateTransformStats` listener AND that the state that is saved with that listener is `STOPPED` for both the indexer and the task. 

I am not sure we want to change the state of the task and the indexer to STOPPED inside memory, but I do think it should be persisted as such, and then call shutdown.